sensor/kubernetes/listener/resources/convert.go

```diff
@@ -461,7 +461,7 @@ func (w *deploymentWrap) updatePortExposureFromServices(svcs ...serviceWithRoute
 }
 
 func (w *deploymentWrap) updatePortExposure(svc serviceWithRoutes) {
-	if svc.selector.Matches(labels.Set(w.PodLabels)) {
+	if svc.selector.Matches(createLabelsWithLen(labels.Set(w.PodLabels))) {
```

Comment:
I think the conversion here is redundant.
```suggestion
	if svc.selector.Matches(createLabelsWithLen(w.PodLabels)) {
```

Reply:
Good point, I'll fix this